internal/relationtuple/manager_requirements.go

```diff
@@ -309,6 +328,36 @@ func ManagerTest(t *testing.T, m Manager, addNamespace func(context.Context, *te
 			require.NoError(t, err)
 			assert.Equal(t, []*InternalRelationTuple{rs[1], rs[3]}, res)
 		})
+
+		t.Run("case=tuple and subject namespace differ", func(t *testing.T) {
```

Comment:
So it is possible to have a relation in a different namespace and link it to a subject in another namespace? Would that cause issues when we delete something something in namespace `2` on which something in namespace `3` depends on?

Reply:
Yes, this is by design. With the new schema we can actually query for those cases as well when deleting, but I am not sure if it makes much sense to actually delete them; ex:
```
groups:admins#member@aeneas
files:/foo#delete@(groups:admins#member)
```
If we delete the first tuple, I don't think you want to delete the second one, even though currently there is no admin.

The de-serialized table structure now actually allows us to query e.g. for tuples referring to nothing, but you might actually want to keep those data anyway.

Reply:
Could there be a case where you remove a relation, later recreate that relation with the same subject ID (but it refers to someone else / new), and now that new subject ID has access to something it shouldn't have (because the old ID had access to it but was not cleared properly)?

Reply:
Yes, but that is why it is so important to have unique IDs everywhere.

Reply:
Can't we design the system to be resilient against something like that as opposed to hoping that people apply it correctly?

Reply:
We can, but that requires us to define the subject IDs and it would remove a lot of the flexibility we have with the relation tuple approach.
I would rather solve this on an API level where we add a way to delete including references, see #599

Reply:
That makes sense, also as per our discussion in the office